src/rez/package_order.py

```diff
@@ -354,73 +489,84 @@ def __init__(self, timestamp, rank=0):
             rank (int): If non-zero, allow version changes at this rank or above
                 past the timestamp.
         """
+        super().__init__(packages)
         self.timestamp = timestamp
         self.rank = rank
 
-    def reorder(self, iterable, key=None):
+        # dictionary mapping from package family to the first-version-after
+        # the given timestamp
+        self._cached_first_after = {}
+        self._cached_sort_key = {}
+
+    def _get_first_after(self, package_family):
+        """Get the first package version that is after the timestamp"""
+        first_after = self._cached_first_after.get(package_family, KeyError)
+        if first_after is KeyError:
+            first_after = self._calc_first_after(package_family)
+            self._cached_first_after[package_family] = first_after
+        return first_after
+
+    def _calc_first_after(self, package_family):
+        from rez.packages import iter_packages
```

Comment:
Was there a reason to put that import here instead of at the top of the module?

Reply:
Not a good one.